packages/ckeditor5-mention/src/mentionediting.ts

```diff
@@ -149,43 +163,41 @@ function createViewMentionElement( mention, { writer } ) {
 	return writer.createAttributeElement( 'span', attributes, options );
 }
 
-// Model post-fixer that disallows typing with selection when the selection is placed after the text node with the mention attribute or
-// before a text node with mention attribute.
-//
-// @param {module:engine/model/writer~Writer} writer
-// @param {module:engine/model/document~Document} doc
-// @returns {Boolean} Returns `true` if the selection was fixed.
-function selectionMentionAttributePostFixer( writer, doc ) {
+/**
+ * Model post-fixer that disallows typing with selection when the selection is placed after the text node with the mention attribute or
+ * before a text node with mention attribute.
+ */
+function selectionMentionAttributePostFixer( writer: Writer, doc: Document ) {
 	const selection = doc.selection;
 	const focus = selection.focus;
 
-	if ( selection.isCollapsed && selection.hasAttribute( 'mention' ) && shouldNotTypeWithMentionAt( focus ) ) {
+	if ( selection.isCollapsed && selection.hasAttribute( 'mention' ) && shouldNotTypeWithMentionAt( focus! ) ) {
 		writer.removeSelectionAttribute( 'mention' );
 
 		return true;
 	}
 }
 
-// Helper function to detect if mention attribute should be removed from selection.
-// This check makes only sense if the selection has mention attribute.
-//
-// The mention attribute should be removed from a selection when selection focus is placed:
-// a) after a text node
-// b) the position is at parents start - the selection will set attributes from node after.
-function shouldNotTypeWithMentionAt( position ) {
+/**
+ * Helper function to detect if mention attribute should be removed from selection.
+ * This check makes only sense if the selection has mention attribute.
+ *
+ * The mention attribute should be removed from a selection when selection focus is placed:
+ * a) after a text node
+ * b) the position is at parents start - the selection will set attributes from node after.
+ */
+function shouldNotTypeWithMentionAt( position: Position ): boolean {
 	const isAtStart = position.isAtStart;
 	const isAfterAMention = position.nodeBefore && position.nodeBefore.is( '$text' );
 
 	return isAfterAMention || isAtStart;
 }
 
-// Model post-fixer that removes the mention attribute from the modified text node.
-//
-// @param {module:engine/model/writer~Writer} writer
-// @param {module:engine/model/document~Document} doc
-// @returns {Boolean} Returns `true` if the selection was fixed.
-function removePartialMentionPostFixer( writer, doc, schema ) {
-	const changes = doc.differ.getChanges();
+/**
+ * Model post-fixer that removes the mention attribute from the modified text node.
+ */
+function removePartialMentionPostFixer( writer: Writer, doc: Document, schema: Schema ): boolean {
+	const changes = doc.differ.getChanges() as Array<DiffItemInsert | DiffItemRemove>;
```

Comment:
Instead of the cast, add on the beginning of `for` loop:

```typescript
if ( change.type == 'attribute' ) {
	continue;
}
```

The coverage shouldn't drop.